src/statement-list/statement_list_typecheck.cpp

```diff
@@ -905,14 +1074,28 @@ void statement_list_typecheckt::typecheck_statement_list_accu_real_arith(
   }
 }
 
-void statement_list_typecheckt::typecheck_instruction_with_operand(
+void statement_list_typecheckt::typecheck_accumulator_compare_instruction(
+  const irep_idt &comparison)
+{
+  const exprt &accu1{accumulator.back()};
+  const exprt &accu2{accumulator.at(accumulator.size() - 2)};
+  // STL behaviour: ACCU2 is lhs, ACCU1 is rhs.
+  const binary_relation_exprt operation{accu2, comparison, accu1};
+  rlo_bit = operation;
+}
+
+const symbol_exprt *const
```

Comment:
The return value of this function should never be `nullptr`. Thus return a reference instead of a pointer.